core/src/com/unciv/logic/civilization/QuestManager.kt

```diff
@@ -339,6 +437,66 @@ class QuestManager {
 
         return null
     }
+
+    /**
+     * Returns a Great Person [BaseUnit] that is not owned by both the [challenger] and the [civInfo]
+     */
+    private fun getGreatPersonForQuest(challenger: CivilizationInfo): BaseUnit? {
+        val ruleSet = civInfo.gameInfo.ruleSet
+        val greatPeople = ruleSet.units.values
+                .filter { baseUnit -> baseUnit.uniques.any { it.equalsPlaceholderText("Great Person - []") } }
+                .map { it.getThisOrReplacedBaseUnit(ruleSet) }.distinct()
+                .subtract(
+                        challenger.getCivGreatPeople().map { it.baseUnit.getThisOrReplacedBaseUnit(ruleSet) }.distinct().toList())
+                .subtract(
+                        civInfo.getCivGreatPeople().map { it.baseUnit.getThisOrReplacedBaseUnit(ruleSet) }.distinct().toList())
+
+        if (greatPeople.isNotEmpty())
+            return greatPeople.random()
+
+        return null
+    }
+
+    /**
+     * Returns a random [CivilizationInfo] (city state) as a target to be killed by [challenger] as
```

Comment:
Is this really how it is in Civ V? They randomly kill each other off through players? That's harsh, man.